modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -0,0 +1,145 @@
+
+    /** Map with outgoing snapshots. */
+    private final ConcurrentMap<UUID, OutgoingSnapshot> outgoingSnapshots = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param messagingService Messaging service.
+     */
+    public OutgoingSnapshotsManager(MessagingService messagingService) {
+        this.messagingService = messagingService;
+    }
+
+    /**
+     * Returns a messagins service.
+     */
+    public MessagingService messagingService() {
+        return messagingService;
+    }
+
+    @Override
+    public void start() {
+        messagingService.addMessageHandler(TableMessageGroup.class, this::messageHandler);
+    }
+
+    @Override
+    public void stop() throws Exception {
+    }
+
+    /**
+     * Registers an outgoing snapshot in the manager.
+     *
+     * @param snapshotId Snapshot id.
+     * @param outgoingSnapshot Outgoing snapshot.
+     */
+    void startOutgoingSnapshot(UUID snapshotId, OutgoingSnapshot outgoingSnapshot) {
+        outgoingSnapshots.put(snapshotId, outgoingSnapshot);
+    }
+
+    /**
+     * Removes an outgoing snapshot from the manager.
+     *
+     * @param snapshotId Snapshot id.
+     */
+    void stopOutgoingSnapshot(UUID snapshotId) {
+        outgoingSnapshots.remove(snapshotId);
+    }
+
+    private void messageHandler(NetworkMessage networkMessage, NetworkAddress sender, @Nullable Long correlationId) {
+        // Ignore all messages that we can't handle.
+        if (!(networkMessage instanceof SnapshotRequestMessage)) {
+            return;
+        }
+
+        assert correlationId != null;
+
+        OutgoingSnapshot outgoingSnapshot = outgoingSnapshots.get(((SnapshotRequestMessage) networkMessage).id());
+
+        if (outgoingSnapshot == null) {
+            if (LOG.isWarnEnabled()) {
+                LOG.warn("Unexpected snapshot request message has been received [message={}]", networkMessage.toString());
+            }
+
+            return;
+        }
+
+        CompletableFuture<? extends NetworkMessage> responseFuture;
+
+        switch (networkMessage.messageType()) {
+            case TableMessageGroup.SNAPSHOT_META_REQUEST:
+                responseFuture = outgoingSnapshot.handleSnapshotMetaRequest((SnapshotMetaRequest) networkMessage);
+
+                break;
+
+            case TableMessageGroup.SNAPSHOT_MV_DATA_REQUEST:
+                responseFuture = outgoingSnapshot.handleSnapshotMvDataRequest((SnapshotMvDataRequest) networkMessage);
+
+                break;
+
+            case TableMessageGroup.SNAPSHOT_TX_DATA_REQUEST:
+                responseFuture = outgoingSnapshot.handleSnapshotTxDataRequest((SnapshotTxDataRequest) networkMessage);
+
+                break;
+
+            default:
+                return;
+        }
+
+        responseFuture.whenComplete((response, throwable) -> respond(response, throwable, sender, correlationId));
+    }
+
+    private CompletableFuture<Void> respond(
```

Comment:
What's the point of this method? Do we need a TODO here?

Reply:
The point is to implement it properly in the future. Component might be stopped, recipient may already be offline, and so on

Reply:
Let's add a TODO then